### PR TITLE
Add authentication to admin

### PR DESCRIPTION
**Pull request by knittingarch**
This PR introduces simple authentication.  It allows a user to create an account, login, and log out.  Presently, there is no validation on the email field, but that will be corrected in a future PR.

<details>
    <summary>Screenshots</summary>

Screen Shot 2020-11-06 at 4 08 11 PM
<br><br>
Screen Shot 2020-11-06 at 4 08 21 PM
<br><br>
Screen Shot 2020-11-06 at 4 08 30 PM
<br><br>
Screen Shot 2020-11-06 at 4 08 36 PM
</details>

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,11 @@
 Rails.application.routes.draw do
   resources :categories, only: [:show]
   resources :posts, only: [:show]
+  resources :users, only: [:create, :new]
+
+  get "login", to: "sessions#new"
+  post "login", to: "sessions#create"
+  get "logout", to: "sessions#destroy"
```
Comment (danbee) [inline config/routes.rb:8]: What do you think about using a resource route for the session instead?

```ruby
resources :session, only: [:new, :create, :destroy]
```

File: app/views/home/index.html.erb [hunk 2]
```diff
@@ -9,3 +9,15 @@
...
+  <div>
+    <h4>You are logged in with <%= current_user.email %></h4>
+    <%= link_to t("auth.logout"), logout_path %>
+  </div>
+<% else %>
+  <div>
+    <%= link_to t("auth.login"), login_path %> or
+    <%= link_to t("auth.create_account"), new_user_path %>
+  </div>
+<% end %>
```
Comment (danbee) [inline app/views/home/index.html.erb:23]: This feels like a good candidate for a partial. What do you think?

File: Gemfile [hunk 3]
```diff
@@ -3,6 +3,7 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby "2.7.2"
 
+gem "bcrypt"
```
Comment (danbee) [inline Gemfile:6]: 🎉

**Review by danbee — APPROVED** [review]
This looks like a solid implementation!

File: config/routes.rb (hunk 1 above)
Comment (knittingarch) [reply config/routes.rb:8]: I wanted the URL to end with `/login` etc, which is why I went with this option.  Is there a way to uses a resource route and override the naming convention?

File: app/views/home/index.html.erb (hunk 2 above)
Comment (knittingarch) [reply app/views/home/index.html.erb:23]: Good idea!  I'm not sure how I'm going to show the logged in state in actuality, but when I have figured that out, I'll make sure to make the code a partial :)

File: config/routes.rb (hunk 1 above)
Comment (danbee) [reply config/routes.rb:8]: Ah no, in that case your custom routes are the way to go. I would suggest making the logout route a `POST` or `DELETE` though.

**Review by danbee — APPROVED** [review]

**Pull request merged**